iphone/cli/commands/_build.js

```diff
@@ -4945,6 +4945,13 @@ iOSBuilder.prototype.processTiSymbols = function processTiSymbols() {
 		if (this.useJSCore) {
 			contents += '\n#define USE_JSCORE_FRAMEWORK';
 		}
+		var infoPlist = this.infoPlist;
+		if (Array.isArray(infoPlist.UIBackgroundModes) && infoPlist.UIBackgroundModes.indexOf('remote-notification') !== -1) {
+			contents += '\n#define USE_TI_SILENTPUSH';
+		}
+		if (Array.isArray(infoPlist.UIBackgroundModes) && infoPlist.UIBackgroundModes.indexOf('fetch') !== -1) {
+			contents += '\n#define USE_TI_FETCH';
+		}
```

Comment:
This will only add these #defines if you're building for simulator or including all ti modules. That means device and dist builds will never have these set. These new lines should probably be moved after this if/else block and before the `if (!destExists || contents !== fs.readFileSync(dest).toString()) {` line.